frontend/src/components/assessments/assessment-status/EditStatusModals/PublishModal.tsx

```diff
@@ -1,43 +1,20 @@
 import React from "react";
-import { useMutation } from "@apollo/client";
 
-import { PUBLISH_TEST } from "../../../../APIClients/mutations/TestMutations";
-import GET_ALL_TESTS from "../../../../APIClients/queries/TestQueries";
 import Modal from "../../../common/Modal";
-import Toast from "../../../common/Toast";
 
 interface PublishModalProps {
   isOpen: boolean;
   onClose: () => void;
-  assessmentId: string;
+  publishAssessment: () => void;
 }
 
 const PublishModal = ({
   isOpen,
   onClose,
-  assessmentId,
+  publishAssessment,
 }: PublishModalProps): React.ReactElement => {
-  const [publishAssessment, { error }] = useMutation<{
```

Comment:
is there a reason we moved this logic to the button? can we also keep the other files consistent (move delete modal logic to delete button)